### PR TITLE
Parse and publish hashtags in notes when posting

### PR DESCRIPTION
**Pull request by joshuatbrown**
## Issues covered
https://github.com/verse-pbc/issues/issues/44

## Description
Parses hashtags from posts and includes them as actual hashtags as defined in [NIP-24](https://github.com/nostr-protocol/nips/blob/master/24.md).

## How to test
1. Tap the Post button
2. Type some text with a hashtag (like `#photography`)
3. Post the note
4. Inspect the source for the note and find the `t: photography` in it.

## Screenshots/Video
Screenshot 2024-11-18 at 2 32 13 PM

### CODE REVIEW

**Comment by mplorentz** [comment]
👀

File: CHANGELOG.md [hunk 1]
```diff
@@ -11,14 +11,14 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
 ### Release Notes
+- Nos now publishes the hashtags it finds in your note when you post. This means it works the way you’ve always expected it to work. [#44](https://github.com/verse-pbc/issues/issues/44)
 - Fixed galleries expanding past the width of the screen when there are lots of links or images. [#24](https://github.com/verse-pbc/issues/issues/24)
 - Added support for user setting and displaying pronouns.
 - Added display of website urls for user profiles.
 
 ## [1.0.2] - 2024-11-26Z
 
 ### Release Notes
-- Fix typo in minimum age warning
```
Comment (mplorentz) [inline CHANGELOG.md:21]: Why did this go away?

**Review by mplorentz — COMMENTED** [review]
I love that we have this feature now! One request and one thought:
- NIP-27 says that hashtags must be lowercased. I think if someone inputs `#HaShTaG` we need to add it to the tags as `hashtag`.
- I'm not sure how hashtags work elsewhere, but I don't like that they are only terminated with whitespace or the end of the string. I expected punctuation to also terminate them. For example, I expected `check out my #hashtag!` to add the hashtag `hashtag`. I'm flexible on this one.

File: CHANGELOG.md (hunk 1 above)
Comment (joshuatbrown) [reply CHANGELOG.md:21]: If I had to guess, I'd say it was a merge issue after we released 1.0.2, but I don't know for sure. I'll figure it out.

Comment (joshuatbrown) [reply CHANGELOG.md]: Restored!

**Comment by joshuatbrown** [comment]
@mplorentz thanks for the request and thought! I've implemented both, so it's ready for another review.

**Review by mplorentz — APPROVED** [review]
Awesome, thanks!

**Pull request merged**